packages/superset-ui-chart-controls/src/operators/utils/getMetricOffsetsMap.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable camelcase */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitationsxw
+ * under the License.
+ */
+import { getMetricLabel, ensureIsArray } from '@superset-ui/core';
+import { PostProcessingFactory } from '../types';
+import { TIME_COMPARISON_SEPARATOR } from './constants';
+
+export const getMetricOffsetsMap: PostProcessingFactory<Map<string, string>> = (
+  formData,
+  queryObject,
+) => {
+  /*
+    return metric offset-label and metric-label hashmap, for instance:
+    {
+      "SUM(value)__1 year ago": "SUM(value)",
+      "SUM(value)__2 year ago": "SUM(value)"
+    }
+  */
+  const queryMetrics = ensureIsArray(queryObject.metrics);
+  const timeOffsets = ensureIsArray(formData.time_compare);
+
+  const metricLabels = queryMetrics.map(getMetricLabel);
+  const metricOffsetMap = new Map<string, string>();
+  metricLabels.forEach((metric: string) => {
+    timeOffsets.forEach((offset: string) => {
+      metricOffsetMap.set([metric, offset].join(TIME_COMPARISON_SEPARATOR), metric);
+    });
+  });
+
+  return metricOffsetMap;
+};
```

Comment:
This may be overkill, but if we standardize how these types of generated labels are rendered, we could then parse them, similarly to how we do with the forecasts, where we know that `col` is just a regular column, `col__yhat` is a point estimate, `col__yhat_lower` is the lower confidence interval etc. In this case maybe we could do `col__offset__1 year ago` in which case we could easily regex this into parts where we know that:
- the column name is `col`
- the derived type of column is `offset`
- the offset is `1 year ago`

etc.

Reply:
For these labels, we should have a mechanism to allow users to customize. I think that we should use a separate PR to design how to customize the derived column label.
Including:
- Preventing label duplication
- Preventing label and column duplication
- How to customize labels in the frontend